cmd/test/cmd.go

```diff
@@ -156,42 +157,73 @@ func doRun(ctx context.Context, options *Options, ioCtrl *io.Controller, k8sLogg
-	}
-	if err = c.Run(ctx, &deployCMD.Options{
-		Manifest:         manifest,
-		ManifestPathFlag: options.ManifestPathFlag,
-		ManifestPath:     options.ManifestPath,
-		Name:             options.Name,
-		Namespace:        options.Namespace,
-		K8sContext:       options.K8sContext,
-		Variables:        options.Variables,
-		Build:            false,
-		Dependencies:     false,
-		Timeout:          options.Timeout,
-		RunWithoutBash:   false,
-		RunInRemote:      manifest.Deploy.Remote,
-		Wait:             true,
-		ShowCTA:          false,
-	}); err != nil {
-		oktetoLog.Error("deploy failed: %s", err.Error())
-		return err
+	kubeClient, _, err := k8sClientProvider.Provide(okteto.GetContext().Cfg)
+	if err != nil {
+		return fmt.Errorf("could not instantiate kuberentes client: %w", err)
+	}
+
+	namer := deployCMD.Namer{
+		KubeClient:   kubeClient,
+		Workdir:      initialCWD,
+		ManifestPath: options.ManifestPathFlag,
+		ManifestName: options.Name,
+	}
+
+	name := options.Name
+	if name == "" {
+		name = namer.ResolveName(ctx)
+	}
+
+	namespace := manifest.Namespace
+	if namespace == "" {
+		namespace = okteto.GetContext().Namespace
+	}
+
+	if !pipeline.IsDeployed(ctx, name, namespace, kubeClient) {
+		c := deployCMD.Command{
+			GetManifest: func(path string, fs afero.Fs) (*model.Manifest, error) {
+				return manifest, nil
+			},
+			K8sClientProvider:  k8sClientProvider,
+			Builder:            builder,
+			GetDeployer:        deployCMD.GetDeployer,
+			EndpointGetter:     deployCMD.NewEndpointGetter,
+			DeployWaiter:       deployCMD.NewDeployWaiter(k8sClientProvider, k8sLogger),
+			CfgMapHandler:      configmapHandler,
+			Fs:                 fs,
+			PipelineCMD:        pc,
+			AnalyticsTracker:   at,
+			IoCtrl:             ioCtrl,
+			K8sLogger:          k8sLogger,
+			IsRemote:           env.LoadBoolean(constants.OktetoDeployRemote),
+			RunningInInstaller: config.RunningInInstaller(),
+		}
+		if err = c.Run(ctx, &deployCMD.Options{
+			Manifest:         manifest,
+			ManifestPathFlag: options.ManifestPathFlag,
+			ManifestPath:     options.ManifestPath,
+			Name:             options.Name,
+			Namespace:        options.Namespace,
+			K8sContext:       options.K8sContext,
+			Variables:        options.Variables,
+			Build:            false,
+			Dependencies:     false,
+			Timeout:          options.Timeout,
+			RunWithoutBash:   false,
+			RunInRemote:      manifest.Deploy.Remote,
+			Wait:             true,
+			ShowCTA:          false,
+		}); err != nil {
+			oktetoLog.Error("deploy failed: %s", err.Error())
+			return err
+		}
+	} else {
+		// The deploy operation expands environment variables in the manifest. If
+		// we don't deploy, make sure to expand the envvars
+		if err := manifest.ExpandEnvVars(); err != nil {
+			return fmt.Errorf("failed to expand manifest environment variables.: %w", err)
+		}
+		oktetoLog.Information("'%s' was already deployed. To redeploy run 'okteto deploy'", name)
```

Comment:
Should we include also `okteto test --deploy` as we do with `okteto up` [here](https://github.com/okteto/okteto/blob/07bd7bb3505aeea89c6735e5f07a7d5552162d69/cmd/up/up.go#L307)?

Reply:
For now that flag doesn't exist. We should probably add it but I'll leave that for a future PR